src/api.rs

```diff
@@ -589,32 +627,39 @@ impl<T: Pixel> Context<T> {
     Ok(())
   }
 
+  /// Returns the details of a specific frame
   pub fn get_frame(&self, frame_number: u64) -> Arc<Frame<T>> {
     // Clones only the arc, so low cost overhead
     self.frame_q.get(&frame_number).as_ref().unwrap().as_ref().unwrap().clone()
   }
 
+  /// Returns the frame count
   pub fn get_frame_count(&self) -> u64 {
     self.frame_count
   }
 
+  /// Set the limit which is passed as argument
   pub fn set_limit(&mut self, limit: u64) {
     self.limit = limit;
   }
 
+  /// Allow to read frames ahead of current ones
   pub fn needs_more_lookahead(&self) -> bool {
     self.needs_more_frames(self.frame_count) && self.frames_processed + LOOKAHEAD_FRAMES > self.frame_q.keys().last().cloned().unwrap_or(0)
   }
 
+  /// Request for more frames
   pub fn needs_more_frames(&self, frame_count: u64) -> bool {
     self.limit == 0 || frame_count < self.limit
   }
 
+  /// Video sequences are headed by a sequence header which was starting value, various other things like chroma sampling x and y, sequence bit depth
```

Comment:
You should document what the function returns. Bonus point if you add a reference to the AV1 specification.